R/forecast.R

```diff
@@ -99,48 +107,105 @@ as_forecast_generic <- function(data,
 }
 
 
-#' @rdname as_forecast
+#' @title Create a `forecast` object for binary forecasts
+#' @description
+#' Create a `forecast` object for binary forecasts. See more information on
+#' forecast types and expected input formats by calling `?`[as_forecast()].
 #' @export
+#' @inheritParams as_forecast
+#' @seealso [as_forecast()], [as_forecast_point()], [as_forecast_binary()],
+#'   [as_forecast_sample()], [as_forecast_quantile()]
 #' @importFrom cli cli_warn
-as_forecast_point <- function(data,
-                              forecast_unit = NULL,
-                              observed = NULL,
-                              predicted = NULL,
-                              model = NULL) {
+#' @keywords check-forecasts
+as_forecast_binary <- function(data,
+                               forecast_unit = NULL,
+                               observed = NULL,
+                               predicted = NULL,
+                               model = NULL) {
   data <- as_forecast_generic(data, forecast_unit, observed, predicted, model)
-  data <- new_forecast(data, "forecast_point")
+  data <- new_forecast(data, "forecast_binary")
   assert_forecast(data)
   return(data)
 }
 
 
-#' @rdname as_forecast
+#' @title Create a `forecast` object for point forecasts
+#' @description
+#' Create a `forecast` object for point forecasts. See more information on
+#' forecast types and expected input formats by calling `?`[as_forecast()].
+#' @inherit as_forecast params
+#' @param ... Unused
+#' @seealso [as_forecast()], [as_forecast_point()], [as_forecast_binary()],
+#'   [as_forecast_sample()], [as_forecast_quantile()]
+#' @export
+#' @keywords check-forecasts
+as_forecast_point <- function(data, ...) {
+  UseMethod("as_forecast_point")
+}
+
+
+#' @rdname as_forecast_point
 #' @export
 #' @importFrom cli cli_warn
-as_forecast_binary <- function(data,
-                               forecast_unit = NULL,
-                               observed = NULL,
-                               predicted = NULL,
-                               model = NULL) {
+as_forecast_point.default <- function(data,
+                                      forecast_unit = NULL,
+                                      observed = NULL,
+                                      predicted = NULL,
+                                      model = NULL,
+                                      ...) {
   data <- as_forecast_generic(data, forecast_unit, observed, predicted, model)
-  data <- new_forecast(data, "forecast_binary")
+  data <- new_forecast(data, "forecast_point")
   assert_forecast(data)
   return(data)
 }
 
 
-#' @rdname as_forecast
+#' @rdname as_forecast_point
+#' @description
+#' When converting a `forecast_quantile` object into a `forecast_point` object,
+#' the 0.5 quantile is extracted and returned as the point forecast.
+#' @export
+#' @keywords check-forecasts
+as_forecast_point.forecast_quantile <- function(data, ...) {
```

Comment:
I thought specific s3 methods *had* to have all the arguments of the default? This could be a argument for a specific data frame method potentially.

Reply:
I think it just has to have the same args as the main function
```
as_forecast_point <- function(data, ...) {
  UseMethod("as_forecast_point")
}
```
At least checks are passing

Reply:
ah right not default. no that makes sense